src/Lotto.js

```diff
@@ -1,18 +1,117 @@
+const { LOTTO_DETAILS, ERROR, LOTTO } = require("./constant/constant");
+const { Console } = require("@woowacourse/mission-utils");
+const Validation = require("./utils/Validation");
+const UI = require("./utils/UI");
+const { RANK } = LOTTO_DETAILS;
+
 class Lotto {
   #numbers;
 
-  constructor(numbers) {
+  constructor(numbers, userLottoArr) {
     this.validate(numbers);
-    this.#numbers = numbers;
+    this.getBonusNumber(userLottoArr);
   }
 
   validate(numbers) {
-    if (numbers.length !== 6) {
-      throw new Error("[ERROR] 로또 번호는 6개여야 합니다.");
+    const splitNumbers = numbers.split(",");
+    Validation.isNumber(splitNumbers.join(""));
+    Validation.isLottoInput(splitNumbers);
+    Validation.isOverlap(splitNumbers);
+    splitNumbers.forEach((lottoNumber) => {
+      Validation.isLottoVariable(Number(lottoNumber));
+    });
+
+    const toNumberSplitNumbers = splitNumbers.map((string) => Number(string));
+    this.#numbers = toNumberSplitNumbers;
+  }
+
+  validateBonusNumber(bonusNumber) {
+    Validation.isNumber(bonusNumber);
+    const toNumberBonusNumber = Number(bonusNumber);
+    Validation.isIncludeNumberInArr(bonusNumber, this.#numbers);
+    Validation.isLottoVariable(toNumberBonusNumber);
+  }
+
+  getBonusNumber(userLottoArr) {
+    Console.readLine(LOTTO.BONUS_ANNOUNCEMENT, (bonusNumber) => {
+      const toNumberBonusNumber = Number(bonusNumber);
+      this.validateBonusNumber(toNumberBonusNumber);
+      const winningBoard = this.calculate(toNumberBonusNumber, userLottoArr);
+      const profit = this.getProfit(winningBoard, userLottoArr);
+      this.printWinningBoard(winningBoard, profit);
+    });
+  }
+
+  calculate(bonusNumber, userLottoArr) {
+    const winningBoard = { ...LOTTO.WINNING_BOARD };
+
+    userLottoArr.forEach((lotto) => {
+      const intersectionLottoAndUser = lotto.filter((x) =>
+        this.#numbers.includes(x)
+      ).length;
```

Comment:
filter를 사용하고 length로 길이를 반환받는것 보다 reduce를 쓰면 한번에 개수를 반환받을 수 있어서 좋은 것 같아요😎
```suggestion
      const intersectionLottoAndUser = lotto.reduce((acc, number) => {
        if(this.#numbers.includes(number)) acc += 1;
        return acc;
        }, 0);
```